### PR TITLE
Implements the Verification System

### LINKED ISSUE #446: Add temporary ability to verify resources
In order to have a successful v2 launch of PHLask, we need to make sure that resources on the platform are verified. Currently, each resource in our database has a `verified` field, which is set to `false` by default. 

We would like a way to easily mark resources as verified. Rather than building out a dashboard for admins (which we will do eventually), we will implement a feature within the app that allows us to mark a resource as verified or as not valid (and therefore deleted). The idea is that team members can go check various resources in person (or in streetview) and easily mark the resource as verified. 

The below screenshots show a proposed design / flow for how this would work. Here are the main ideas:

- A tag showing "VERIFIED" or "UNVERIFIED" is added to the detailed view of a resource. In the images below, I found that the bottom right of the image header of the resource is a good spot.
- When that tag is clicked, a dialog pops up. An admin can enter a password (which we will provide to PHLask members) and then choose whether this should be marked as verified or deleted.
- Based on what the user chooses, a call to firebase is made - either to delete that resource, or update its verified field to true.

image

Note that this feature may be removed soon after the launch, as we build out more official verification tools.

### PR DESCRIPTION
**Pull request by vontell**
Implements a temporary verification system that allows users to verify and un-verify resources. Please message a PHLASK convener for the password.

Also does the following:

* Attaches an `id` to resources within the UI for mapping back to Firebase.
* Updates the regex for website submissions to allow for more website types.
* Updates the verification data schema type.

Screenshot 2024-06-22 at 4 02 05 PM
Screenshot 2024-06-22 at 4 03 30 PM
Screenshot 2024-06-22 at 4 03 37 PM

### CODE REVIEW

File: src/actions/actions.js [hunk 1]
```diff
@@ -54,6 +61,13 @@ export const pushNewResource = newResource => ({
   newResource
 });
 
+// Handles the case where an existing resource is updated from the submission form
+export const UPDATE_EXISTING_RESOURCE = 'UPDATE_EXISTING_RESOURCE';
+export const updateExistingResource = updatedResource => ({
```
Comment (RNR1) [inline src/actions/actions.js]:
```suggestion
export const updateExistingResource = createAction('UPDATE_EXISTING_RESOURCE');
```

File: src/reducers/filterMarkers.js [hunk 2]
```diff
@@ -97,6 +97,16 @@ export default (state = initialState, act) => {
         allResources: [...state.allResources, act.newResource]
       };
 
+    case actions.UPDATE_EXISTING_RESOURCE:
```
Comment (RNR1) [inline src/reducers/filterMarkers.js]:
```suggestion
    case actions.updateExistingResource.type:
      allResources: state.allResources.map(resource =>
          resource.id === act.payload.id
            ? act.payload
            : resource
        )
```

File: src/components/Verification/VerificationButton.jsx [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+                    CANCEL
+                  </Button>
+                </div>
+              )}
+              {hasBeenUpdated && (
+                <div>
+                  <p>Your change has been recorded. Thanks!</p>
+                  <Button
+                    onClick={closeModal}
+                    style={{
```
Comment (RNR1) [inline src/components/Verification/VerificationButton.jsx]: use `sx` with MUI components

File: src/components/Verification/VerificationButton.jsx [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+                      backgroundColor: 'grey',
+                      color: 'white',
+                      padding: '10px',
+                      margin: '5px',
+                      borderRadius: '5px',
+                      width: '100%'
+                    }}
+                  >
+                    CLOSE
```
Comment (RNR1) [inline src/components/Verification/VerificationButton.jsx]: nit: MUI buttons by default are uppercase - but I'm not sure if we overridden this in the theme

File: src/components/Verification/VerificationButton.jsx [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+                    style={{
+                      backgroundColor: 'grey',
+                      color: 'white',
+                      padding: '10px',
+                      margin: '5px',
+                      borderRadius: '5px',
+                      width: '100%'
+                    }}
+                  >
+                    CANCEL
```
Comment (RNR1) [inline src/components/Verification/VerificationButton.jsx]: nit: MUI buttons by default are uppercase - but I'm not sure if we overridden this in the theme

File: src/components/Verification/VerificationButton.jsx [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+                      margin: '5px',
+                      borderRadius: '5px',
+                      width: '100%'
+                    }}
+                  >
+                    MARK AS INACTIVE
+                  </Button>
+                  <Button
+                    onClick={closeModal}
+                    style={{
```
Comment (RNR1) [inline src/components/Verification/VerificationButton.jsx]: use `sx` for MUI components

File: src/components/Verification/VerificationButton.jsx [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+                      backgroundColor: 'red',
+                      color: 'white',
+                      padding: '10px',
+                      margin: '5px',
+                      borderRadius: '5px',
+                      width: '100%'
+                    }}
+                  >
+                    MARK AS INACTIVE
```
Comment (RNR1) [inline src/components/Verification/VerificationButton.jsx]: nit: MUI buttons by default are uppercase - but I'm not sure if we overridden this in the theme

File: src/components/Verification/VerificationButton.jsx [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+                    style={{
+                      backgroundColor: 'orange',
+                      color: 'white',
+                      padding: '10px',
+                      margin: '5px',
+                      borderRadius: '5px',
+                      width: '100%'
+                    }}
+                  >
+                    MARK AS UNVERIFIED
```
Comment (RNR1) [inline src/components/Verification/VerificationButton.jsx]: nit: MUI buttons by default are uppercase - but I'm not sure if we overridden this in the theme

File: src/components/Verification/VerificationButton.jsx [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+                  </Button>
+                  <Button
+                    onClick={markAsInactive}
+                    style={{
```
Comment (RNR1) [inline src/components/Verification/VerificationButton.jsx]: sx for MUI components

File: src/components/Verification/VerificationButton.jsx [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+                type="text"
+                value={name}
+                onChange={e => {
+                  setName(e.target.value);
+                }}
+              />
+              <Input
+                placeholder="The admin password"
+                style={{ width: '100%', marginBottom: '10px' }}
+                type="text"
```
Comment (RNR1) [inline src/components/Verification/VerificationButton.jsx]: nit: use type="password"

File: src/components/Verification/VerificationButton.jsx [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+                    color: 'white',
+                    borderRadius: '4px',
+                    padding: '4px'
+                  }}
+                >
+                  Cancel
+                </Button>
+                <Button
+                  onClick={() => {
+                    if (btoa(password) == PASSWORD) {
```
Comment (RNR1) [inline src/components/Verification/VerificationButton.jsx]: nit: use `===`

**Review by RNR1 — COMMENTED** [review]
First pass!

File: src/actions/actions.js (hunk 1 above)
Comment (vontell) [reply src/actions/actions.js]: Fixed

File: src/components/Verification/VerificationButton.jsx (hunk 10 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Fixed.

File: src/components/Verification/VerificationButton.jsx (hunk 11 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Good catch, fixed!

File: src/components/Verification/VerificationButton.jsx (hunk 8 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Got it - fixed.

File: src/components/Verification/VerificationButton.jsx (hunk 9 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Fixed

File: src/components/Verification/VerificationButton.jsx (hunk 7 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Fixed

File: src/components/Verification/VerificationButton.jsx (hunk 6 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Fixed

File: src/components/Verification/VerificationButton.jsx (hunk 5 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Fixed

File: src/components/Verification/VerificationButton.jsx (hunk 3 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Fixed

File: src/components/Verification/VerificationButton.jsx (hunk 4 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Fixed

File: src/reducers/filterMarkers.js (hunk 2 above)
Comment (vontell) [reply src/reducers/filterMarkers.js]: Fixed

**Review by vontell — COMMENTED** [review]
@RNR1 back to you!

File: src/actions/actions.js [hunk 12]
```diff
@@ -48,6 +53,10 @@ export const pushNewResource = newResource => ({
   newResource
 });
 
+// Handles the case where an existing resource is updated from the submission form
+export const UPDATE_EXISTING_RESOURCE = 'UPDATE_EXISTING_RESOURCE';
```
Comment (RNR1) [inline src/actions/actions.js]: The action type is not necessary anymore! You can use 
```js
updateExistingResource.type
```

File: src/components/Verification/VerificationButton.jsx [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+        >
+          {!isAdmin && (
+            <div>
+              <p>
+                Enter the admin password to update the verified status of this
+                resource. Also enter your name.
+              </p>
+              <Input
+                placeholder="Your name"
+                style={{ width: '100%', marginBottom: '10px' }}
```
Comment (gcardonag) [inline src/components/Verification/VerificationButton.jsx]: Similar to other mui components, we should use the `sx` attribute here in place of `style`

File: src/components/Verification/VerificationButton.jsx [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+                type="text"
+                autoComplete="disabled"
+                value={name}
+                onChange={e => {
+                  setName(e.target.value);
+                }}
+              />
+              <Input
+                placeholder="The admin password"
+                style={{ width: '100%', marginBottom: '10px' }}
```
Comment (gcardonag) [inline src/components/Verification/VerificationButton.jsx]: Similar to other mui components, we should use the `sx` attribute here in place of `style`

File: src/components/Verification/VerificationButton.jsx [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+              >
+                <Button
+                  onClick={closeModal}
+                  style={{
+                    flex: 1,
+                    backgroundColor: 'gray',
+                    color: 'white',
+                    borderRadius: '4px',
+                    padding: '4px'
+                  }}
```
Comment (gcardonag) [inline src/components/Verification/VerificationButton.jsx]: Similar to other mui components, we should use the `sx` attribute here in place of `style`

File: src/components/Verification/VerificationButton.jsx [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+                      setLoginError('Invalid password');
+                    }
+                  }}
+                  style={{
+                    flex: 1,
+                    backgroundColor: 'blue',
+                    color: 'white',
+                    borderRadius: '4px',
+                    padding: '4px'
+                  }}
```
Comment (gcardonag) [inline src/components/Verification/VerificationButton.jsx]: Similar to other mui components, we should use the `sx` attribute here in place of `style`

**Review by gcardonag — CHANGES_REQUESTED** [review]
A few minor changes that should not impact functionality and this will be good to go!

File: src/components/Verification/VerificationButton.jsx (hunk 13 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Fixed!

File: src/components/Verification/VerificationButton.jsx (hunk 14 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Fixed!

File: src/components/Verification/VerificationButton.jsx (hunk 15 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Fixed!

File: src/components/Verification/VerificationButton.jsx (hunk 16 above)
Comment (vontell) [reply src/components/Verification/VerificationButton.jsx]: Fixed!

**Review by gcardonag — APPROVED** [review]
Approved, thanks for making all of those changes! ✅

File: src/actions/actions.js (hunk 12 above)
Comment (vontell) [reply src/actions/actions.js]: Fixed!

**Review by gcardonag — APPROVED** [review]
🧹

**Pull request merged**